pkg/util/kernel/headers.go

```diff
@@ -0,0 +1,157 @@
+// +build linux
+
+package kernel
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strconv"
+
+	"github.com/DataDog/datadog-agent/pkg/metadata/host"
+	"github.com/mholt/archiver/v3"
+)
+
+const sysfsHeadersPath = "/sys/kernel/kheaders.tar.xz"
+const kernelModulesPath = "/lib/modules/%s/build"
+
+func FindHeaderDirs() ([]string, error) {
+	hv, err := HostVersion()
+	if err != nil {
+		return nil, fmt.Errorf("unable to determine host kernel version: %w", err)
+	}
+
+	dir, err := getHeaderDirs(hv)
+	if err == nil {
+		return dir, nil
+	}
+
+	if os.IsNotExist(err) {
+		// as last resort, look for sysfs headers
+		if dir, syserr := getSysfsHeaderDirs(hv); syserr == nil {
```

Comment:
Do we need to worry about removing the temp path this creates?

Reply:
We want it stick around, unless we want to extract the headers on every run on system-probe.

Reply:
I added code to remove it in the case that it fails to validate properly, so we don't attempt to use it again.